src/net.cpp

```diff
@@ -443,94 +454,99 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
+            bool proxyConnectionFailed = false;
 
-        if (addrConnect.IsI2P() && use_proxy) {
-            i2p::Connection conn;
-            bool connected{false};
+            if (target_addr.IsI2P() && use_proxy) {
+                i2p::Connection conn;
+                bool connected{false};
 
-            if (m_i2p_sam_session) {
-                connected = m_i2p_sam_session->Connect(addrConnect, conn, proxyConnectionFailed);
-            } else {
-                {
-                    LOCK(m_unused_i2p_sessions_mutex);
-                    if (m_unused_i2p_sessions.empty()) {
-                        i2p_transient_session =
-                            std::make_unique<i2p::sam::Session>(proxy, &interruptNet);
-                    } else {
-                        i2p_transient_session.swap(m_unused_i2p_sessions.front());
-                        m_unused_i2p_sessions.pop();
+                if (m_i2p_sam_session) {
+                    connected = m_i2p_sam_session->Connect(target_addr, conn, proxyConnectionFailed);
+                } else {
+                    {
+                        LOCK(m_unused_i2p_sessions_mutex);
+                        if (m_unused_i2p_sessions.empty()) {
+                            i2p_transient_session =
+                                std::make_unique<i2p::sam::Session>(proxy, &interruptNet);
+                        } else {
+                            i2p_transient_session.swap(m_unused_i2p_sessions.front());
+                            m_unused_i2p_sessions.pop();
+                        }
                     }
-                }
-                connected = i2p_transient_session->Connect(addrConnect, conn, proxyConnectionFailed);
-                if (!connected) {
-                    LOCK(m_unused_i2p_sessions_mutex);
-                    if (m_unused_i2p_sessions.size() < MAX_UNUSED_I2P_SESSIONS_SIZE) {
-                        m_unused_i2p_sessions.emplace(i2p_transient_session.release());
+                    connected = i2p_transient_session->Connect(target_addr, conn, proxyConnectionFailed);
+                    if (!connected) {
+                        LOCK(m_unused_i2p_sessions_mutex);
+                        if (m_unused_i2p_sessions.size() < MAX_UNUSED_I2P_SESSIONS_SIZE) {
+                            m_unused_i2p_sessions.emplace(i2p_transient_session.release());
+                        }
                     }
                 }
-            }
 
-            if (connected) {
-                sock = std::move(conn.sock);
-                addr_bind = CAddress{conn.me, NODE_NONE};
+                if (connected) {
+                    sock = std::move(conn.sock);
+                    addr_bind = CAddress{conn.me, NODE_NONE};
+                }
+            } else if (use_proxy) {
+                LogPrintLevel(BCLog::PROXY, BCLog::Level::Debug, "Using proxy: %s to connect to %s:%s\n", proxy.ToString(), target_addr.ToStringAddr(), target_addr.GetPort());
```

Comment:
nit: since this line is being changed anyway, consider this:

```suggestion
                LogPrintLevel(BCLog::PROXY, BCLog::Level::Debug, "Using proxy: %s to connect to %s\n", proxy.ToString(), target_addr.ToStringAddrPort());
```

This would take care to put IPv6 addresses in `[` `]` instead of printing something like `2001:470:88ff:2e::1:8333`

Reply:
This is just being moved but agreed is simple enough to include it here.

Covered in [fd81a37](https://github.com/bitcoin/bitcoin/pull/28834/commits/fd81a37239541d0d508402cd4eeb28f38128c1f2)